File: Matlab_PF/create_dc.m
function dcNet = create_dc(case_name)
% CREATE_DC Reads DC grid data from CSV files.
%
%   dcNet = CREATE_DC(case_name) reads the DC grid data from CSV files
%   with the given case_name prefix. The following CSV files are required:
%       case_name_baseMW_dc.csv
%       case_name_pol_dc.csv
%       case_name_bus_dc.csv
%       case_name_branch_dc.csv
%       case_name_conv_dc.csv
%
%   The function returns a structure dcNet with the fields:
%       dcNet.baseMW    - Scalar base MW value 
%       dcNet.pol       - Scalar pol value 
%       dcNet.bus       - DC bus data matrix
%       dcNet.branch    - DC branch data matrix
%       dcNet.converter - DC converter data matrix

    base_path = pwd;
    
    %---------------------------List .csv files----------------------------
    required_files = { ...
        [case_name '_baseMW_dc.csv'], ...
        [case_name '_pol_dc.csv'], ...
        [case_name '_bus_dc.csv'], ...
        [case_name '_branch_dc.csv'], ...
        [case_name '_conv_dc.csv'] ...
    };

    %-------------------------Check missing files--------------------------
    missing_files = {};
    for i = 1:length(required_files)
        file_path = fullfile(base_path, required_files{i});
        if ~isfile(file_path)
            missing_files{end+1} = required_files{i};
        end
    end
    if ~isempty(missing_files)
        error('Missing required DC files: %s', strjoin(missing_files, ', '));
    end

    %------------------------Read .csv file data---------------------------
    dcNet.baseMW = readmatrix(fullfile(base_path, ...
        [case_name '_baseMW_dc.csv']));

    dcNet.pol = readmatrix(fullfile(base_path, ...
        [case_name '_pol_dc.csv']));

    dcNet.bus = readmatrix(fullfile(base_path, ...
        [case_name '_bus_dc.csv']));

    dcNet.branch = readmatrix(fullfile(base_path, ...
        [case_name '_branch_dc.csv']));

    dcNet.converter = readmatrix(fullfile(base_path, ...
        [case_name '_conv_dc.csv']));
end

File: OPF_julia/create_dc.jl
using CSV
using DataFrames

"""
    create_dc(case_name::String) -> Dict{String,Any}

Load DC grid data from CSV files with the given `case_name` prefix.

This function expects the following files to be present in the same directory as this source file:
  - `<case_name>_baseMW_dc.csv`
  - `<case_name>_pol_dc.csv`
  - `<case_name>_bus_dc.csv`
  - `<case_name>_branch_dc.csv`
  - `<case_name>_conv_dc.csv`

Returns a dictionary `dc` with the following keys:
  - "baseMW"    : Scalar base MW value.
  - "pol"       : Scalar pole value.
  - "bus"       : Matrix containing bus data.
  - "branch"    : Matrix containing branch data.
  - "converter" : Matrix containing converter data.
"""
function create_dc(case_name::String)
    # Set the base path to the directory containing this file.
    base_path = @__DIR__
    
    # Initialize the dictionary to store DC grid data.
    dc = Dict{String,Any}()

    # List of required CSV files.
    required_files = [
        "$(case_name)_baseMW_dc.csv",
        "$(case_name)_pol_dc.csv",
        "$(case_name)_bus_dc.csv",
        "$(case_name)_branch_dc.csv",
        "$(case_name)_conv_dc.csv"
    ]
    
    # Check for missing files.
    missing_files = filter(f -> !isfile(joinpath(base_path, f)), required_files)
    if !isempty(missing_files)
        error("Missing required DC files: " * join(missing_files, ", "))
    end
    
    # Load each CSV file
    read_csv_matrix(file) = Matrix(DataFrame(CSV.File(joinpath(base_path, file); header=false)))
    
    baseMW = read_csv_matrix("$(case_name)_baseMW_dc.csv")
    dc["baseMW"] = baseMW[1, 1]
    
    pol = read_csv_matrix("$(case_name)_pol_dc.csv")
    dc["pol"] = pol[1, 1]
    
    dc["bus"]       = read_csv_matrix("$(case_name)_bus_dc.csv")
    dc["branch"]    = read_csv_matrix("$(case_name)_branch_dc.csv")
    dc["converter"] = read_csv_matrix("$(case_name)_conv_dc.csv")
    
    return dc
end

